joss/brachistochrone.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import openmdao.api as om
+import dymos as dm
+import matplotlib.pyplot as plt
+
+# First define a system which computes the equations of motion
+class BrachistochroneEOM(om.ExplicitComponent):
+    def initialize(self):
+        self.options.declare('num_nodes', types=int)
+
+    def setup(self):
+        nn = self.options['num_nodes']
+
+        # Inputs
+        self.add_input('v', val=np.zeros(nn), units='m/s', desc='velocity')
+        self.add_input('theta', val=np.zeros(nn), units='rad', desc='angle of wire')
+        self.add_output('xdot', val=np.zeros(nn), units='m/s', desc='x rate of change')
+        self.add_output('ydot', val=np.zeros(nn), units='m/s', desc='y rate of change')
+        self.add_output('vdot', val=np.zeros(nn), units='m/s**2', desc='v rate of change')
+
+        # Ask OpenMDAO to compute the partial derivatives using complex-step
+        # with a partial coloring algorithm for improved performance
+        self.declare_partials(of='*', wrt='*', method='cs')
+        self.declare_coloring(wrt='*', method='cs', show_summary=True)
+
+    def compute(self, inputs, outputs):
+        v, theta = inputs.values()
```

Comment:
Just thinking. The purpose of this new file is to provide a reference of the file used in the JOSS paper, so that the reader can download dymos and see the work?

I don't think we show this full-vector method anywhere in the OpenMDAO examples, do we? And we haven't made an effort to start using it (since it is supposed to be more efficient.)  Maybe we need a story to convert over to using it more often in OpenMDAO (and in dymos too.)

Reply:
I think we should include examples which CS through single-component ODEs.  We don't need to do that with all cases, but Kaushik's update of SSTO and the new balanced field example both demonstrate it.

Reply:
Oh, what I meant was this: ` v, theta = inputs.values()` .  We don't teach that (yet), i don't think.